Meadowlark-js/packages/meadowlark-core/src/validation/DocumentReferenceExtractor.ts

```diff
@@ -76,6 +76,29 @@ const multiZip = (...theArrays) => {
   return firstArray.map((value, index) => restArrays.reduce((acc, array) => [...acc, array[index]], [value]));
 };
 
+/**
+ * Recursively collects JSON document paths from sub-ReferenceComponents, which represent identities that are
+ * references to other entities. In a MetaEd model, there is no limit to the number of entities that reference others
+ * as a part of their identity.
+ */
+function documentPathsFromSubReferenceComponent(
+  subReferenceComponent: ReferenceComponent,
+  referenceComponentTopLevelName: string,
+  result: string[],
+) {
```

Comment:
Trivial change requested: I like functions better when they return any modified references, so that it is more clear to the consumer that the reference has in fact been modified. For that reason in C# I usually reject the use of `ref` or `out` parameters. What do you think about changing this to return the `result` object?

I think Uncle Bob has a term for this in Clean Code... avoid side-effects, I think.